### ISSUE TITLE
Example: Controlling Wifi/Guestwifi

### ISSUE THREAD

**Issue by Lynnaut** [Aypac/Arduino-TR-064-SOAP-Library#3]
Hey,

First of all a big thanks to you for providing the community with this awesome library.

I want to turn on/off my WiFi especially my GuestWiFi using an esp8266.
But I do not really understand how to put a fitting command together to use this feature.
In your ToDo you mention that you want to add this scenario to the example sketches.
Do you got this working? And if so could you provide me with some example code?

Greetings Tim

**Comment by Aypac** [comment]
Hi Tim.
that should be pretty easy to implement. The hard part is finding the correct function call, which is usually not very intuitive (and not at all well documented!). I guess it should be somewhere in here: https://192.168.179.1:49443/wlanconfigSCPD.xml

(You have to call this address from within your network and confirm the certificate exception).

My guess is, that you just have to enable the second WiFi, for which the call should look something like:
```
String params[][2] = {{"NewEnable", 1}}; //Maybe 1, maybe true, maybe "enable", who knows what the value here should be...
String req[][2] = {{"Enable", ""},}; //should be the return value
connection.action("urn:dslforum-org:service:WLANConfiguration:2", "SetEnable", params, 1, req, 2);
USE_SERIAL.println("WiFi enabled " + (req[1][1]));
```
The 2 in "urn:dslforum-org:service:WLANConfiguration:2" should indicate, that it is the guest WiFi (not sure, maybe you also have more WiFi's? Walk through https://192.168.179.1:49443/tr64desc.xml to find out) and "SetEnable" is the action/function you call with the parameter {"NewEnable", 1} and return value "Enable". Unfortunately I don't have the time right now to test around, but please let me know how it went. If you can't get anywhere, I might check it out one day.

Good Luck,
René

P.S.: For what it is worth, the interface is documented  [here](https://avm.de/fileadmin/user_upload/Global/Service/Schnittstellen/AVM_TR-064_first_steps.pdf) and [here](https://www.broadband-forum.org/technical/download/TR-064.pdf).

**Comment by gickowtf** [comment]
Hey guys here my Work for a toggle the GuestWLAN:

`#include <Arduino.h>
#if defined(ESP8266)
  #include <ESP8266WiFi.h>
  #include <ESP8266WiFiMulti.h>
  #include <ESP8266HTTPClient.h>
  ESP8266WiFiMulti WiFiMulti;
#elif defined(ESP32)
  #include <WiFi.h>
  #include <WiFiMulti.h>
  #include <HTTPClient.h>
  WiFiMulti WiFiMulti;
#endif

#include <tr064.h>
const byte taster = D3;
const byte led = D5;
bool flag, swich, state;

//-------------------------------------------------------------------------------------
// Put your router settings here
//-------------------------------------------------------------------------------------


// Wifi network name (SSID)
const char* wifi_ssid = "++++++++++++++"; 

// Wifi network password
const char* wifi_password = "+++++++++++++++++++++";

// The username and password for the API of your router
// The username if you created and account, "admin" otherwise
// ("Anmeldung aus dem Heimnetz mit Benutzername und Passwort")
const char* fuser = "++++++++++";
const char* fpass = "+++++++++++++";

// IP address of your router. This should be "192.168.178.1" for most FRITZ!Boxes
const char* IP = "192.168.178.1";

// Port of the API of your router. This should be 49000 for all TR-064 devices.
const int PORT = 49000;

// -------------------------------------------------------------------------------------

// Do not mess with these :)
// TR-064 connection
TR064 connection(PORT, IP, fuser, fpass);


void setup() {
  Serial.begin(115200);
  // Wait a few secs for warm-up (dunno why, was in the default code for http connections).
  delay(5000);
 
  //Connect to wifi
  WiFiMulti.addAP(wifi_ssid, wifi_password);

  // Wait for the wifi to connect
  while ((WiFiMulti.run() != WL_CONNECTED)) {
    delay(100);
  }
  pinMode(taster, INPUT_PULLUP);
  pinMode(led, OUTPUT);
}

void loop() {
  swich = !digitalRead(taster); // Taster einlesen

  if (swich && !flag) {   // pos Flanke erkennen
    flag = true;          // Flankenmerke setzten
    state = !state;       // status toggeln
    if (state)gwlanon();    // Funktion on afrufen
    else gwlanoff();       // Funktion off aufrufe
    delay(50);// entprellen
    }

  if (!swich) {
    flag = false; // reset flankenmerker
    delay(50);    // entprellen
  }
    digitalWrite(led, state);// led an /aus
}

  
    


void gwlanon() {

  // (Re-) Initialize the TR-064 library - it is done every time, as maybe the connection has lost before
  // TODO: This should be fixed with b33fbbc6, we need to check if that did the job.
  connection.init();

  String params[][2] = {{"NewEnable", "1"}};
  String req[][2] = {{}};

  connection.action("urn:dslforum-org:service:WLANConfiguration:3", "SetEnable", params, 1, req, 2);
  Serial.println ("Status ein");
}

void gwlanoff() {

  // (Re-) Initialize the TR-064 library - it is done every time, as maybe the connection has lost before
  // TODO: This should be fixed with b33fbbc6, we need to check if that did the job.
  connection.init();

  String params[][2] = {{"NewEnable", "0"}};
  String req[][2] = {{}};

  connection.action("urn:dslforum-org:service:WLANConfiguration:3", "SetEnable", params, 1, req, 2);
  Serial.println("Status aus");
} `

**Comment by Aypac** [comment]
Good work @gickowtf , thank you. I will test, translate, clean up and integrate it later into the examples folder.

Can anyone verify if this runs with their router? Is it always "urn:dslforum-org:service:WLANConfiguration:**3**" or is it sometimes number 1 or 2 (or 4)?

**Comment by gickowtf** [comment]
At my fritzbox is 1 for 2,4 2 for 5 ghz and 3 for guest wlan... sorry for my english

**Comment by Aypac** [comment]
> **Quote of AVM**
    If the device additionally supports a second physical Access Point e.g. FRITZ!Box 7390 support 2.4 GHz and 5 GHz, one more service is listed (service 2).
    If the device supports a logical Access Point for guests, one more service is listed (service 2 or 3).

So yeah, it's either number 2 or 3, depending on wheather or not you have a 5GHz WiFi.

**Comment by Aypac** [comment]
See https://github.com/Aypac/Arduino-TR-064-SOAP-Library/tree/new_examples/examples/WifiSwitch
This needs to be merged into main soon.

**Comment by Aypac** [comment]
Should be implemented/included with  7469e9e (v1.2.2)
https://github.com/Aypac/Arduino-TR-064-SOAP-Library/tree/master/examples/Wifi_Switch
